src/components/advanced/sign_and_verify.vue

```diff
@@ -0,0 +1,220 @@
+<template>
+  <div class="sign-and-verify">
+    <div class="q-pa-md">
+      <div class="q-mb-lg description">
+        {{ $t("strings.signAndVerifyDescription") }}
+      </div>
+      <div class="text-h6">Sign</div>
+      <div class="row justify-between items-end">
+        <LokiField :label="$t('fieldLabels.data')">
+          <q-input
+            v-model.trim="toSign"
+            :dark="theme == 'dark'"
+            borderless
+            dense
+            :placeholder="$t('placeholders.dataToSign')"
+          />
+        </LokiField>
+        <div class="btn-wrapper q-ml-md q-py-sm">
+          <q-btn
+            color="primary"
+            :label="$t('buttons.sign')"
+            :loading="sign_status.sending"
+            :disable="!toSign"
+            @click="sign()"
+          />
+        </div>
+      </div>
+      <div class="verify-heading text-h6">Verify</div>
+      <div class="justify-between items-end">
+        <LokiField class="q-mt-md" :label="$t('fieldLabels.signature')">
+          <q-input
+            v-model.trim="signatureToVerify"
+            :dark="theme == 'dark'"
+            borderless
+            dense
+            :placeholder="$t('placeholders.signature')"
+          />
+        </LokiField>
+        <LokiField class="q-mt-md" :label="$t('fieldLabels.data')">
+          <q-input
+            v-model.trim="unsignedData"
+            :dark="theme == 'dark'"
+            borderless
+            dense
+            :placeholder="$t('placeholders.unsignedData')"
+          />
+        </LokiField>
+        <LokiField class="q-mt-md" :label="$t('fieldLabels.address')">
+          <q-input
+            v-model.trim="address"
+            :dark="theme == 'dark'"
+            borderless
+            dense
+            :placeholder="$t('placeholders.address')"
+          />
+        </LokiField>
+        <div class="submit-button">
+          <q-btn
+            color="primary"
+            :label="$t('buttons.verify')"
+            :disable="!signatureToVerify || !unsignedData || !address"
+            @click="verify()"
+          />
+          <q-btn
+            v-if="canClear"
+            :label="$t('buttons.clear')"
+            color="secondary"
+            @click="clear"
+          />
+        </div>
+        <SignatureDialog
+          :on-copy="copySignature"
+          :on-close="closeDialog"
+          :signature="signature"
+          :show="!!signature"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+const { clipboard } = require("electron");
+import LokiField from "components/loki_field";
+import SignatureDialog from "./signature_dialog";
+import { mapState } from "vuex";
+export default {
+  name: "SignAndVerify",
+  components: {
+    LokiField,
+    SignatureDialog
+  },
+  data() {
+    return {
+      toSign: "",
+      // returned from the RPC
+      signature: "",
```

Comment:
couldn't we use a computed value to watch for this rather than set a local state?
`signature: state.gateway.sign_status.signature`

Reply:
True, I think that can be done

Reply:
fixed